py/call_iterator.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 The go-python Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// CallIterator objects
+
+package py
+
+// A python CallIterator object
+type CallIterator struct {
+	callable Object
+	sentinel Object
+}
+
+var CallIteratorType = NewType("callable_iterator", "callable_iterator type")
+
+// Type of this object
+func (o *CallIterator) Type() *Type {
+	return CallIteratorType
+}
+
+func (cit *CallIterator) M__iter__() (Object, error) {
+	return cit, nil
+}
+
+// Get next one from the iteration
+func (cit *CallIterator) M__next__() (Object, error) {
+	f, ok := cit.callable.(*Function)
```

Comment:
You don't need to check this (and it might be a bad idea since, for instance bound methods can be called).  `py.Call` will check and return the correct TypeError